doc/user/content/concepts/indexes.md

```diff
@@ -13,29 +13,35 @@ aliases:
 ## Overview
 
 In Materialize, indexes represent query results stored in memory **within a
-[cluster](/concepts/clusters/)**. You can create indexes on
-[sources](/concepts/sources/), [views](/concepts/views/#views), or [materialized
+[cluster](/concepts/clusters/)**. Indexes in Materialize store **both** the key
+and any associated rows.
+
+You can create indexes on [sources](/concepts/sources/), tables,
+[views](/concepts/views/#views), or [materialized
 views](/concepts/views/#materialized-views).
 
-## Indexes on sources
+## Indexes on sources and tables
 
 {{< note >}}
 In practice, you may find that you rarely need to index a source
-without performing some transformation using a view, etc.
-{{</ note >}}
+or a table without performing some transformation using a view, etc.
+{{</ note>}}
 
-In Materialize, you can create indexes on a [source](/concepts/sources/) to
-maintain in-memory up-to-date source data within the cluster you create the
-index. This can help improve [query
-performance](#indexes-and-query-optimizations) when serving results directly
-from the source or when [using joins](/transform-data/optimization/#join).
-However, in practice, you may find that you rarely need to index a source
-directly.
+In Materialize, you can create indexes on a [source](/concepts/sources/) or a
+[table](/sql/create-table) to maintain in-memory up-to-date source or table data
+in the cluster you create the index. This can help improve [query
+performance](#indexes-and-query-optimizations) when [using
+joins](/transform-data/optimization/#join) or when serving results directly
+from a source or a table. However, in practice, you may find that you rarely
+need to index a source or a table directly.
 
 ```mzsql
 CREATE INDEX idx_on_my_source ON my_source (...);
+CREATE INDEX idx_on_my_table ON my_table (...);
 ```
 
+Indexes in Materialize store **both** the key and any associated rows.
```

Comment:
We could either 1) include the parallel to other databases for context, or 2) include more context about the implications of storing both the key and the associated rows (_i.e_, make it about the size implications, rather than what the index stores). There's some blurb about the structure and memory footprint of indexes in the [`CREATE INDEX` page](https://materialize.com/docs/sql/create-index/#structure).

Reply:
Will add a little of 1 and 2 context in next patch.